tokio/tests/task_scope.rs

```diff
@@ -0,0 +1,341 @@
+#![warn(rust_2018_idioms)]
+#![cfg(feature = "full")]
+
+use futures::{select, FutureExt};
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    time::{Duration, Instant},
+};
+use tokio::{
+    task::{scope, Scope, ScopeConfigBuilder},
+    time::delay_for,
+};
+
+#[tokio::test]
+async fn unused_scope() {
```

Comment:
What behavior is this testing? That dropping it panics? Should the scoped future even be polled?

Reply:
I guess the idea behind it was to validate that "Futures which aren't polled do nothing". Not too important, but still adds a bit of test coverage 😀

Reply:
Oh ok, so `drop(scope(|_| async { unreachable!() }))` shouldn't hit the unreachable, do I understand correctly?

Reply:
Yes. `scope` returns a `Future` which executes lazily just like any other `Future`